nucliadb_node/src/shards/shard_writer.rs

```diff
@@ -447,8 +476,31 @@ impl ShardWriter {
         text_read(&self.text_writer).count()
     }
 
+    #[tracing::instrument(skip_all)]
+    pub fn merge(&self) -> NodeResult<()> {
+        let current = self.writes_since_merge.swap(0, Ordering::SeqCst);
+
+        if current < MERGE_THRESHOLD {
+            self.writes_since_merge.fetch_add(current, Ordering::SeqCst);
+            return Ok(());
+        }
```

Comment:
If swapping with a 0 and then adding current, don't you get the same value?

Reply:
Only if there were not enough writes to be merge worthy, I add instead of store because I dont want to assume concurrency mechanisms in this code. Therefore `writes_since_merge` may have change

Reply:
Yes, but what I mean is the following. Imagine no concurrent merges:
- You enter and `writes_since_merge` is 3
- After the swap is 0 and `current` is 3
- If `current < MERGE_THRESHOLD` you add `current` to 0, so you have 3 again and return

You have not incremented the counter right?

Reply:
> If current < MERGE_THRESHOLD you add current to 0, so you have 3 again and return

Why would I increment the counter? I did not perform a write on the shard, therefore I leave the state of `writes_since_merge` as I found it (since I did not perform a merge).
Also remember that I dont assume concurrency on top of `ShardWriter` so when I add the value of `writes_since_merge` may not be zero.

Reply:
Oh sorry, I though your intention was to increment there